### PR TITLE
Initiate Login Flow 

### PR DESCRIPTION
**Pull request by rosasck**
Calls keycloak to begin the login flow and acts like a proxy back to the client

### CODE REVIEW

File: definitions.go [hunk 1]
```diff
@@ -695,3 +697,35 @@ type ClientScopeRepresentation struct {
 	ProtocolMappers *[]ProtocolMapperRepresentation `json:"protocolMappers,omitempty"`
 	Attributes      *map[string][]string            `json:"attributes,omitempty"`
 }
+
+type InitatePKCELogin struct {
+	Nonce         string `json:"nonce"`
```
Comment (efabens) [inline definitions.go]: I don't believe that this should have json tags

**Review by efabens — APPROVED** [review]
approve pending tags. I also i am in the process of using a library that should automate this. But want to verify it does what I think it does before sending you down that path.

File: login.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+	"github.com/pkg/errors"
+	"gopkg.in/h2non/gentleman.v2"
+)
+
+// InitiateLogin begins the login flow
+func (c *Client) InitiateLogin(realmName string, loginURLEncoded InitatePKCELogin) (*http.Response, error) {
+	var req *gentleman.Request
+	{
+		var authPath = fmt.Sprintf("/auth/realms/%s/protocol/openid-connect/auth", realmName)
```
Comment (galxy25) [inline login.go]: pull this up like https://github.com/tozny/keycloak-client/blob/master/clients.go#L13

**Review by galxy25 — APPROVED** [review]

File: login.go (hunk 2 above)
Comment (rosasck) [reply login.go]: Since i have to use sprintf, I added it as a const! but I cant URL param replace it like the example in clients

**Comment by rosasck** [comment]
Complete Unless more feedback

**Pull request merged**